staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/objectmeta/coerce.go

```diff
@@ -18,40 +18,78 @@ package objectmeta
 
 import (
 	"fmt"
+	"strings"
 
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/apimachinery/pkg/runtime"
 	utiljson "k8s.io/apimachinery/pkg/util/json"
+	"k8s.io/klog/v2"
 )
 
 // GetObjectMeta does conversion of JSON to ObjectMeta. It first tries json.Unmarshal into a metav1.ObjectMeta
 // type. If that does not work and dropMalformedFields is true, it does field-by-field best-effort conversion
 // throwing away fields which lead to errors.
 func GetObjectMeta(obj map[string]interface{}, dropMalformedFields bool) (*metav1.ObjectMeta, bool, error) {
+	meta, found, _, err := GetObjectMetaStrict(obj, dropMalformedFields, false)
+	return meta, found, err
+}
+
+// GetObjectMetaStrict allows for strict unmarshaling of the metadata bytes, returning any unknown field paths
+// it encounters
+func GetObjectMetaStrict(obj map[string]interface{}, dropMalformedFields, strict bool) (*metav1.ObjectMeta, bool, []string, error) {
```

Comment:
it's sort of weird to have a GetObjectMetaStrict function where strict is a boolean parameter

would a GetObjectMetaWithOptions function that takes an ObjectMetaOptions{DropMalformedFields, ReturnUnknownPaths} be clearer?